.github/workflows/build.yml

```diff
@@ -35,6 +35,17 @@ jobs:
       - name: Run unit tests
         run: make test
       - name: Run e2e tests
-        run:  make e2e
+        run:  |
+          #!/bin/sh
+          if [[ $GITHUB_REF_NAME == v* && $GITHUB_REF_TYPE == tag ]]; then
+            make e2e
+          else
+            export GOCOVERDIR=$GITHUB_WORKSPACE/test/e2e/.cover
```

Comment:
Is it better to define the `GOCOVERDIR` in Makefile, so the make `make e2e-covdata` can work locally?

Reply:
added another dedicated make target to run e2e and generate coverage data